### DISCUSSION TITLE
UEFI Windows Install error "A required device isn't connected or can't be accessed"

### DISCUSSION THREAD

**Discussion by Lucho1976** [ipxe/ipxe#1070] [Q&A] [answered]
Hi, guys, I have an issue with a Windows installation that is working correctly for BIOS but not for UEFI. As you can see in the screenshots, everything related to UEFI seems to be loading fine, but towards the end, I get the message: "A required device isn't connected or can't be accessed."
Any idea what might be happening?
Thanks in advance!

![foro1](https://github.com/ipxe/ipxe/assets/26300768/0545acca-bf29-4584-a02b-10ea66733137)

![foro2](https://github.com/ipxe/ipxe/assets/26300768/470ae67d-3e4e-40e9-a11c-a1831c700cc8)

![BootEfi2](https://github.com/ipxe/ipxe/assets/26300768/fc67ad99-9298-4640-9699-8f464ce3cd61)

**boot.ipxe**
#!ipxe

set server_ip 192.168.0.180

kernel tftp://${server_ip}/images/win2019/wimboot/wimboot
initrd tftp://${server_ip}/images/win2019/boot/bcd bcd
initrd tftp://${server_ip}/images/win2019/boot/boot.sdi boot.sdi
initrd tftp://${server_ip}/images/win2019/sources/WinPE.wim boot.wim
boot

**Comment by NiKiZe** [comment, marked as answer]
You do not need BCD or boot.sdi with latest versions of wimboot.
Also try using a default boot.wim from Windows installation media.

Reading more carefully, your .wim is loaded as WinPE.wim (see your screenshot) the reason for this is that in efi mode you need the -n argument, while for pcbios the name is given after.

    initrd -n boot.wim tftp://${server_ip}/images/win2019/sources/WinPE.wim boot.wim

Would probably work, however your screenshot is using http while your script says tftp, something is of here ;)

**Reply by Lucho1976** [reply]
You are a genius NiKiZe, you saved my day. That worked fine. Good catch about http and ftp, one is my local and the other production (sorry). I will try without BCD and boot.sdi, thanks a lot!
